src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -2,40 +2,64 @@
 import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
 import ChannelList from '../../../database/schema/ChannelListSchema';
 import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useDatabaseQueueHook from '../queue/useDatabaseQueueHook';
 import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import StorageUtils from '../../../utils/storage';
 
 const useFetchPostNotificationHook = () => {
   const {localDb, refresh} = useLocalDatabaseHook();
+  const {queue} = useDatabaseQueueHook();
 
   const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
     if (!localDb) return;
-    const allPromises: Promise<void>[] = notifications.map((notification) => {
-      const channelList = fromNotificationAPI(notification);
-      return channelList.save(localDb);
+
+    notifications.map((notification) => {
+      queue.addJob({
+        label: 'save post notifications',
+        task: async () => {
+          const channelList = fromNotificationAPI(notification);
+          await channelList.save(localDb);
+        }
+      });
+
+      return null;
     });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `map` function is used here, but the return value is not used. It would be more appropriate to use `forEach` instead of `map` if you're not using the returned array.
```diff
- notifications.map((notification) => {
+ notifications.forEach((notification) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->